docs/src/man/basics.md

```diff
@@ -0,0 +1,1220 @@
+([67, 22, 49, 45, 53, 35, 53, 35, 61, 28  …  37, 34, 23, 30, 50, 31, 40, 38, 23, 27], [67, 22, 49, 45, 53, 35, 53, 35, 61, 28  …  37, 34, 23, 30, 50, 31, 40, 38, 23, 27], [67, 22, 49, 45, 53, 35, 53, 35, 61, 28  …  37, 34, 23, 30, 50, 31, 40, 38, 23, 27])
+
+julia> german."Sex", german[!, "Sex"] # the same using string indexing
+(["male", "female", "male", "male", "male", "male", "male", "male", "male", "male"  …  "male", "male", "male", "male", "male", "female", "male", "male", "male", "male"], ["male", "female", "male", "male", "male", "male", "male", "male", "male", "male"  …  "male", "male", "male", "male", "male", "female", "male", "male", "male", "male"])
+
+julia> german[:, 3] # note that this creates a copy
+1000-element PooledArrays.PooledVector{String, UInt32, Vector{UInt32}}:
+ "male"
+ "female"
+ "male"
+ "male"
+ "male"
+ "male"
+ "male"
+ "male"
+ "male"
+ "male"
+ ⋮
+ "male"
+ "male"
+ "male"
+ "male"
+ "female"
+ "male"
+ "male"
+ "male"
+ "male"
+```
+
+To grab one row as DataFrame, we can index as follows:
+```jldoctest dataframe
+julia> german[2:2, :]
+1×10 DataFrame
+ Row │ id     Age    Sex     Job    Housing  Saving accounts  Checking account ⋯
+     │ Int64  Int64  String  Int64  String   String           String           ⋯
+─────┼──────────────────────────────────────────────────────────────────────────
+   1 │     1     22  female      2  own      little           moderate         ⋯
+                                                               3 columns omitted
+
+julia> german[3, :] # this produces a DataFrameRow which is treated as 1-dimensional object similar to a NamedTuple
+DataFrameRow
+ Row │ id     Age    Sex     Job    Housing  Saving accounts  Checking account ⋯
+     │ Int64  Int64  String  Int64  String   String           String           ⋯
+─────┼──────────────────────────────────────────────────────────────────────────
+   3 │     2     49  male        1  own      little           NA               ⋯
+                                                               3 columns omitted
+```
+
+we can grab a single cell or element with the same syntax to grab an element of an array:
+```jldoctest dataframe
+julia> german[4, 4]
+2
+```
+
+or to get a new DataFrame that is subset of rows and columns:
+```jldoctest dataframe
+julia> german[4:5, 4:5]
+2×2 DataFrame
+ Row │ Job    Housing
+     │ Int64  String
+─────┼────────────────
+   1 │     2  free
+   2 │     2  free
+```
+
+## Not, Between, Cols, and All selectors
+
+Finally, you can use `Not`, `Between`, `Cols`, and `All` selectors in more complex column selection
+scenarioes (note that `Cols()` selects no columns while `All()` selects all columns therefore `Cols`
+is a preferred selector if you write generic code). The following examples move all columns whose
+names match `r"x"` regular expression respectively to the front and the end of a dataframe. Now we will
```

Comment:
do not use regexp examples before you introduced the user to the fact that it can be used.